src/app/Http/Controllers/VolunteerController.php

```diff
@@ -284,6 +294,37 @@ public function store(Request $request) {
             }
             /** Add the 'courses' to the volunteer. */
             $volunteer->courses = $data['courses'];
+        }else{
+            /** if is specified only a course */
+            $course_name_id =  $request->has('course_name_id') ? $request->course_name_id : '';
+            if (trim($course_name_id)!='') {
+                $course_name = CourseName::find($course_name_id);
+                if ($course_name) {
+                    /** Create a new course. */
+                    $newCourse = [
+                        'course_name' => [
+                            '_id' => $course_name['_id'],
+                            'name' => $course_name['name'],
+                            'slug' => removeDiacritics($course_name['name'])],
+                        'obtained' => Carbon::parse($request->get('obtained',''))->format('Y-m-d H:i:s')
+                    ];
+                    /** Check if the accreditor already exists in DB. */
+                    $courseAccreditor = CourseAccreditor::query()->where('name', '=', $request->get('accredited_by',''))->first();
+                    if (!$courseAccreditor) {
+                        $courseAccreditor = CourseAccreditor::create(['name' => $request->get('accredited_by',''), 'courses' => [$course_name['_id']]]);
+                    } else {
+                        if (is_array($courseAccreditor->courses) && !in_array($course_name['_id'], $courseAccreditor->courses)) {
+                            $courseAccreditor->courses = array_merge( $courseAccreditor->courses, [$course_name['_id']]);
+                            $courseAccreditor->save();
+                        }
+                    }
+                    /** Save the accreditor and add the course to the volunteer. */
+                    $newCourse['accredited'] = ['_id' => $courseAccreditor->_id,'name' => $courseAccreditor->name];
+                    $data['courses'][] = $newCourse;
+                    $volunteer->courses = $data['courses'];
+                }
+            }
+
```

Comment:
Have you tested this?
So in case only one course was added to a volunteer, that course was ignored?